### ISSUE TITLE
SelectRecordsInputStream.read() hangs randomly

### ISSUE THREAD

**Issue by manojkumarvohra** [aws/aws-sdk-java#2229]
I am using S3Select to read multiple S3 objects in parallel via different threads in a MapReduce job.

Number of objects: ~ 1000
Avg object Size: ~ 650MB
Format: CSV

**Problem**
Randomly we are seeing that 98% of the objects are queried successfully and towards end of the job few threads get stuck awaiting response @
`SelectRecordsInputStream.read()`

**LIBS**
SDK Version: 1.11.629
HTTPClient: 4.5.9
HTTPCore: 4.4.11

**Workaround**
We are manually setting a timeout on our mapper JVMs so that if they don't see any progress update in configured time they timeout.
On task re-attempt the pending CSV files are picked and processed successfully.
[See attached job progress image]

**Solutions-Tried-So-far**
I tried increasing number of connections in S3Client to 1000, modulated socket timeout, tried both HTTPS and HTTP protocols but nothing helped so far. 

**Client Configuration**
Even though AmazonS3Client is marked @threadsafe but  before querying each S3 object we create a fresh S3Client. We shutdown S3Client post query operation.
We are setting max connections per client to 50 (Default), where each client just querying a single S3 object.

`ClientConfiguration clientConfig = new ClientConfigurationFactory().getConfig()
				.withMaxConnections(50);`

`
this.s3Client = (AmazonS3Client) AmazonS3ClientBuilder.standard().withRegion(this.clientRegion)
				.withCredentials(
						new AWSStaticCredentialsProvider(new BasicAWSCredentials(this.accessKey, this.secretKey)))
				.withClientConfiguration(clientConfig).build();
`

Please guide how we can solve this.

<img width="487" alt="sieve-progress-issue" src="https://user-images.githubusercontent.com/17252390/74586454-7d95fd00-500d-11ea-864a-a303d53834fc.png">

**Comment by debora-ito** [comment]
Can you share a sample code and the stacktrace of the error?

**Comment by debora-ito** [comment]
Closing due to no response, we need more info to help in the investigation.
Feel free to reopen if you have more questions.

**Comment by manojkumarvohra** [comment]
Apologies for no response.
There has been actually no error trace because there is no error at all as the socket kept hanging for data. Only error trace is the timeout error which we force trigger on yarn container.

I tracked this down with AWS premium support.

This is the response I received from them after having round of discussions and sample data sharing.

----------------------
Hello Manoj,   

Thank you for your patience.  

I just received an update from the service team, they found there was a host in the service grid experienced a hardware problem during some of your longest query, which had been removed from the service. 
-------------------------------
